iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -92,6 +92,26 @@ inline expected<ValueType, ErrorType>::expected(error<ErrorType>&& errorValue) n
 {
 }
 
+template <typename ValueType, typename ErrorType>
+inline expected<ValueType, ErrorType>::expected(expected<ValueType, ErrorType>&& rhs) noexcept
+{
+    *this = std::move(rhs);
```

Comment:
I always see this and it is suboptimal in general: the move (copy) ctor is implemented in terms of the (copy) move in terms of the corresponding operator=. Looks fine on first sight from code reuse point of view etc. The problem is that this way in many cases a potentially costly default initialization happens before the assignment.

From my point of view it is often better to do initialize directly with the exception of classes with primitive type members only (not entirely the case here, due to m_store). 

Those inefficiencies add up in the long run.

Maybe consider this in future implementations.